### ISSUE TITLE
Soap funcs with != 1 parameter not working

### ISSUE THREAD

**Issue by Kernle32DLL** [fiorix/wsdl2go#74]
I realized today, that this lib shows odd behavior when using soap funcs which either don't take any parameters, or **more than one**.

The first one of course is explained by [this TODO](https://github.com/fiorix/wsdl2go/blob/master/wsdlgo/encoder.go#L586) in-code. However, the if clause at this position also has an oversight which causes wrong method templating on soap funcs with more than one parameter. In this case the parameters end up in the go func definition, but are not passed to p.cli.RoundTripWithAction.

If welcome, I can cook up at least some test cases, and maybe some potential enhancements to fix the issues.

Edit: I see the "more than 1 parameter" issue is tracked by https://github.com/fiorix/wsdl2go/issues/65

**Comment by fiorix** [comment]
This issue is probably better than #65 to track this. Will close the other one.

**Comment by Kernle32DLL** [comment]
I'm currently working on a fix for zero parameters, and got that working for the most part I think. I'm left to some spec checking tho, as I still encounter some errors.

**Comment by fiorix** [comment]
Awesome, thanks! 👍

**Comment by Kernle32DLL** [comment]
Well, the zero parameter functions are working now. I was suffering from rpc issues as described in #65 .

Current progress can be tracked here: https://github.com/Kernle32DLL/wsdl2go

I got most things working. However, response mapping with RPC is not working yet, as it is difficult to guess the right XML attribute for mapping (not as straight forward as with request). Also, I did not figure out a sane way to initialize potentially complex pointer structures for unmarshaling yet (this is a issue for both rpc **and** non-rpc). Non-rpc should work as before, but this will probably break eventually, as non-rpc was not implemented to spec (as far as I can tell - I might be wrong tho). Non rpc keeps generating a `<message>` xml wrapper tag, which I cannot seem to get rid of - hints welcome.

So - much to do, but making progress nicely.
